bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlValue.java

```diff
@@ -51,6 +52,7 @@
 public abstract class XmlValue implements RefValue, BXml, CollectionValue {
 
     Type type = PredefinedTypes.TYPE_XML;
+    protected TypedescValue typedesc = new TypedescValueImpl(type);
```

Comment:
IMO behavior typeof operation with immutable xml is not covered here . Shall we call the value utils to create the typedesc here.